### PR TITLE
Orca ray estimator

### PR DESCRIPTION
**Pull request by cyita**

### CODE REVIEW

**Comment by jason-dai** [comment]
I think we should have the same base class for ray- and spark-based estimators, so as to ensure consistent interfaces.

**Comment by cyita** [comment]
Jenkins passed http://10.239.47.210:18888/job/ZOO-PR-Validation/4796/

File: pyzoo/zoo/orca/learn/tf2/tf_ray_estimator.py [hunk 1]
```diff
@@ -393,6 +394,15 @@ def restore(self, checkpoint):
...
         ray.get([worker.set_state.remote(state_id) for worker in self.remote_workers])
 
+    def load(self, checkpoint):
+        """Restores the model from the provided checkpoint.
+
+        Args:
+            checkpoint (str): Path to target checkpoint file.
+
+        """
+        self.restore(checkpoint=checkpoint)
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/tf2/tf_ray_estimator.py:404]: why do still need `restore`?

**Comment by jason-dai** [comment]
A basic question is how to unify the interface between ray-based and spark-based estimator APIs. Currently they still have different interfaces and functionalities.

**Comment by cyita** [comment]
Jenkins passed http://10.239.47.210:18888/job/ZOO-PR-Validation/4817/

File: pyzoo/zoo/orca/learn/ray_estimator.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    @abstractmethod
+    def shutdown(self, **kwargs):
+        pass
+
+    @abstractmethod
+    def clear_gradient_clipping(self):
+        pass
+
+    @abstractmethod
+    def set_constant_gradient_clipping(self, min, max):
```
Comment (yangw1234) [inline pyzoo/zoo/orca/learn/ray_estimator.py:54]: Does it make sense to add `set_gradient_clipping` here? Because for some models, like Keras, it is hard to change the gradient clipping behavior after the model has been created. @jason-dai 

Keras gradient clipping code.
```python
model = tf.keras.Model(...)
model.compile(tf.keras.optimizers.Adam(clip_norm=...))
```

File: pyzoo/zoo/orca/learn/base_estimator.py [hunk 3]
```diff
@@ -41,3 +41,27 @@ def save(self, model_path):
...
+    def get_train_summary(self, tag=None):
+        raise NotImplementedError
+
+    @abstractmethod
+    def get_validation_summary(self, tag=None):
+        raise NotImplementedError
+
+    @abstractmethod
+    def set_tensorboard(self, log_dir, app_name):
+        raise NotImplementedError
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/base_estimator.py:67]: Move these to SparkEstimator?

File: pyzoo/zoo/orca/learn/pytorch/estimator.py [hunk 4]
```diff
@@ -83,7 +63,7 @@ def from_torch(*,
                              f" for now, got backend: {backend}")
 
 
-class PyTorchRayEstimatorWrapper(Estimator):
+class PyTorchRayEstimatorWrapper(OrcaRayEstimator):
```
Comment (jason-dai) [inline pyzoo/zoo/orca/learn/pytorch/estimator.py:66]: Can we directly use `PyTorchRayEstimator`, instead of adding a wrapper?

**Comment by jason-dai** [comment]
Change the following:
- `mxnet_trainer.py` --> `estimator.py`
- `tf_ray_estimator.py` --> `estimator.py`
- `PyTorchRayEstimatorWrapper` --> `PyTorchRayEstimator`
- `tf2.Estimator` --> `tf2.TensorFlow2Estimator`
- `mxnet.estimator` --> `mxnet.MXNetEstimator`

**Comment by jason-dai** [comment]
LGTM

**Comment by cyita** [comment]
Jenkins passed http://10.239.47.210:18888/job/ZOO-PR-Validation/4899/
